com.unity.ml-agents/Runtime/Sensor/SensorShapeValidator.cs

```diff
@@ -27,12 +27,12 @@ public void ValidateSensors(List<ISensor> sensors)
                 // Check for compatibility with the other Agents' Sensors
                 // TODO make sure this only checks once per agent
                 Debug.Assert(m_SensorShapes.Count == sensors.Count, $"Number of Sensors must match. {m_SensorShapes.Count} != {sensors.Count}");
-                for (var i = 0; i < m_SensorShapes.Count; i++)
+                for (var i = 0; i < Mathf.Min(m_SensorShapes.Count, sensors.Count); i++)
```

Comment:
If these differed before, we'd get the assert, but would potentially continue and iterate through the lists, so accessing `sensors[i]` would read past the end of the list.